Finished HackYourTemperature App

dyosorio opened the pull request:
Finished HackYourFuture App

Thank you!

olcarmontzaragoza commented on week3/homework/exercises/1-joke-api/script.js:
```diff
@@ -10,9 +10,17 @@
+    console.log(data.value.joke);
+  } catch (error) {
+    console.log(error);
+  }
```
Perfect!

olcarmontzaragoza commented on week3/homework/exercises/2-authentication/script.js:
```diff
@@ -8,8 +8,20 @@
   // YOUR CODE GOES IN HERE
+  try {
+    const response = await fetch('https://restapiabasicauthe-sandbox.mxapps.io/api/books', {
+      headers: { 'Authorization': 'Basic YWRtaW46aHZnWDhLbFZFYQ==' }
```
You don't need quotes around Authorization. It can just be:

Authorization: 'Basic YWRtaW46aHZnWDhLbFZFYQ=='

dyosorio replied:
This was changed.

olcarmontzaragoza commented on week3/homework/exercises/3-party-time/script.js:
```diff
@@ -9,8 +9,26 @@
-function makeReservation() {
+const fetch = require('node-fetch');
+
+async function makeReservation() {
```
You may want to take a parameter here of the guest object that contains name and numberOfPeople (being defined outside of the function)

dyosorio replied:
This has been changed

olcarmontzaragoza commented on week3/homework/exercises/3-party-time/script.js:
```diff
@@ -9,8 +9,26 @@
+async function makeReservation() {
   // YOUR CODE GOES IN HERE
+  const body = {
+    "name": "John Doe",
```
Here, you also don't need to use quotes for property names, can just be:
name: "John Doe",
etc.

dyosorio replied:
This has been changed.

olcarmontzaragoza commented on week3/homework/exercises/3-party-time/script.js:
```diff
@@ -9,8 +9,26 @@
+  }
 }
 
 makeReservation();
```
The rest is perfect!

olcarmontzaragoza commented on week3/homework/exercises/4-handlebars/script.js:
```diff
@@ -42,4 +30,36 @@ const punchlines = [
+  return cardData;
+}
+
+drawCard();
```
You don't need to call drawCard here.

dyosorio replied:
Updated

olcarmontzaragoza commented on week3/homework/exercises/4-handlebars/script.js:
```diff
@@ -42,4 +30,36 @@ const punchlines = [
+drawCard();
+
+//Create a variable, called card
+const card = `${subjects} is great to ${punchlines}`;
```
I'm confused by this: you want one subject not all the array of subjects. I think it would be better you move the templating code within the drawCard function as that way you can have just one subject and just one punchline.

dyosorio replied:
Big mistake I had here! Thanks for your feedback it's been updated

olcarmontzaragoza commented on week3/homework/exercises/4-handlebars/script.js:
```diff
@@ -42,4 +30,36 @@ const punchlines = [
+ */
+function getRandomElement(array) {
+  // YOUR CODE GOES IN HERE
+  const randomEl = array[Math.floor(Math.random() * array.length)];
```
It seems you only need one line here, don't mind about defining a new constant. 
Change it to this:

return array[Math.floor(Math.random() * array.length)];

dyosorio replied:
This has been updated

olcarmontzaragoza commented on week2/homework/HackYourTemperature/server.js:
```diff
@@ -0,0 +1,45 @@
+const exphbs = require('express-handlebars');
+const bodyParser = require('body-parser');
+const axios = require('axios');
+//const API_KEY = require('./sources/keys.json').API_KEY;
```
Why did you comment this out?

olcarmontzaragoza commented on week2/homework/HackYourTemperature/server.js:
```diff
@@ -0,0 +1,45 @@
+//use handlebars as a template to render the template
+app.get('/', function(req, res){
+    //res.send('hello from backend to frontend!');
+    console.log('running on port 3000');
```
You don't need this line

dyosorio replied:
Updated

olcarmontzaragoza commented on week2/homework/HackYourTemperature/server.js:
```diff
@@ -0,0 +1,45 @@
+
+app.post('/weather', function(req, res){
+  const cityName = req.body.cityName;
+  const API_KEY = "a54b947f262e2c5b975e64258dc64f90";
```
Ah I see where you put the API key. Just for organisational reasons, I'd recommend placing this at the top of the file in case you need it for more than one request method.

dyosorio replied:
Updated

dyosorio replied:
Updated

olcarmontzaragoza commented on week2/homework/HackYourTemperature/server.js:
```diff
@@ -0,0 +1,45 @@
+
+  axios.get(`http://api.openweathermap.org/data/2.5/weather?q=${cityName}&units=metric&APPID=${API_KEY}`)
+  .then((response) => {
+    let temp = response.data.main.temp;
```
You may want to round this with Math.floor()

dyosorio replied:
The result looks much better after I changed this. Thanks for your comments!

olcarmontzaragoza commented on week2/homework/HackYourTemperature/views/index.handlebars:
```diff
@@ -0,0 +1,18 @@
+      <input type="submit" values="SUBMIT" class="btn btn-success btn-block mt-2">
+    </div>
+  </form>
+  <p style="text-align: center; width: 80%; background-color:#dcdde1; font-size: 3vh" class="form-control mx-auto">{{ weatherText }}</p>
```
Perfect! Now as mentioned previously, it is advised not to use inline styles here as it could get messy. I would place these in an external css/scss file.

dyosorio replied:
I agree. I updated, thank you

olcarmontzaragoza reviewed (approved):
Hey Danny, good work this week! There were a few errors that I want to fix in a new commit, these were to do mainly with exercise four about drawing a card (see my comments), placing API keys at the top of your file and a few other minor things. Apart from that, everything else looked great, keep it up for the next module! Best of luck ;)

dyosorio commented:
Hi! @olcarmontzaragoza thank you very much for reviewing all of my homework. I'll make the necessary adjustments and let you know once I finished.

Many thanks

dyosorio commented:
Hi, @olcarmontzaragoza I want to thank you for taking the time to go over my code and provide such detailed feedback. I learned a lot from this homework! 

I addressed all your comments and made the necessary changes. Please find them on two new commits, one for the exercises and another one for the HackYourTemperature app. Thank you

olcarmontzaragoza commented:
> Hi, @olcarmontzaragoza I want to thank you for taking the time to go over my code and provide such detailed feedback. I learned a lot from this homework!
> 
> I addressed all your comments and made the necessary changes. Please find them on two new commits, one for the exercises and another one for the HackYourTemperature app. Thank you

Perfect! I'm glad you appreciated my feedback. Great work making the changes so quickly!
